### ISSUE TITLE
[Regex] Hex character ranges

### ISSUE THREAD

**Issue by michaelblyons** [sublimehq/Packages#270]
### Expected behavior

Escaped hex chars (`\x08`) are recognized as a single entity, and can be ranged to another char.

### Actual behavior

The `\x` is considered an escaped character, and ranges pick the least-significant digit of the hex number and "range" it to the `\` of the next `\xNN`, where the `\` is not recognized as an escape character:

```
[\x08-\x0A]
^  ^    ^^^  keyword.control.set.regexp
 ^^ constant.character.escape.regexp
     ^^^ constant.other.range.regexp
```

### Steps to reproduce

1. Pull up the "Find" bar.
2. Put `[\x08-\x0A]` in it.
3. Either look at the highlighting or check the scopes for each position.

### My environment

* Operating system:
    * [ ] Windows
    * [x] Mac OS
    * [ ] Linux
* Sublime Text Build:
  * branch:
    * [x] stable (though I use some language defs from this repo)
    * [ ] dev
  * number:
    * Build `3103`

**Comment by FichteFoll** [comment]
I think I will borrow this as an issue template for https://github.com/SublimeTextIssues/Core/issues :+1:

**Comment by michaelblyons** [comment]
I stole the issue template from #195

**Comment by wbond** [comment]
This should be fixed by 01fea1594bfbc2054810eedca8bf595355b4838c
